internal/expr/functions/builtins.go

```diff
@@ -652,3 +659,62 @@ func (s *AvgAggregator) Eval(_ *environment.Environment) (types.Value, error) {
 func (s *AvgAggregator) String() string {
 	return s.Fn.String()
 }
+
+// Len represents the len() function.
+// It returns the length of string, array or document.
+// For other types len() returns NULL.
+type Len struct {
+	Expr expr.Expr
+}
+
+// Eval extracts the average value from the given document and returns it.
+func (s *Len) Eval(env *environment.Environment) (types.Value, error) {
+	val, err := s.Expr.Eval(env)
+	if err != nil {
+		return nil, err
+	}
+	var length int
+	switch v := val.V().(type) {
+	case string:
+		length = len(v)
+	case types.Array:
+		arrayLen, err := document.ArrayLength(v)
+		if err != nil {
+			return nil, err
+		}
+		length = arrayLen
+	case types.Document:
+		docLen, err := document.Length(v)
+		if err != nil {
+			return nil, err
+		}
+		length = docLen
+	default:
+		return types.NewNullValue(), nil
+	}
+
+	return types.NewIntegerValue(int64(length)), nil
+}
+
+// IsEqual compares this expression with the other expression and returns
+// true if they are equal.
+func (s *Len) IsEqual(other expr.Expr) bool {
+	if other == nil {
+		return false
+	}
+
+	o, ok := other.(*Len)
+	if !ok {
+		return false
+	}
+
+	return expr.Equal(s.Expr, o.Expr)
+}
+
+func (s *Len) Params() []expr.Expr { return []expr.Expr{s.Expr} }
+
+// String returns the alias if non-zero, otherwise it returns a string representation
```

Comment:
This docstring is outdated, perhaps something like this `String returns the literal representation of len.`

Reply:
Fixed it with your suggestion